src/wallet/spend.cpp

```diff
@@ -969,7 +970,21 @@ static util::Result<CreatedTransactionResult> CreateTransactionInternal(
     for (const auto& coin : selected_coins) {
         txNew.vin.push_back(CTxIn(coin.outpoint, CScript(), nSequence));
     }
-    DiscourageFeeSniping(txNew, rng_fast, wallet.chain(), wallet.GetLastBlockHash(), wallet.GetLastBlockHeight());
+
+    // `DiscourageFeeSniping` may backdate locktime, but we prohibit
+    // it if this transaction spends an unconfirmed output or is an
+    // RBF replacement, in order to avoid leaking information that may
+    // serve as a fingerprint of a wallet being used.
```

Comment:
It'd be helpful if you explained in more detail here what exactly is the fingerprinting we're concerned about. There's a lot of possible fingerprinting attacks and it's not clear what you're referring to exactly here.

Reply:
Thanks! Will do.